server/routes/coordinator.js

```diff
@@ -1,27 +1,61 @@
-const httpError = require('http-errors')
+const { AddStaffResult } = require('../services/involvedStaffService')
+const { firstItem } = require('../utils/utils')
 
 module.exports = function Index({ reportService, involvedStaffService, reviewService, offenderService, systemToken }) {
-  /**
-   * TODO: integrate this into the app properly once we have designs and time.
-   * This is currently a GET to get around CSRF issues and to allow access via a browser using current authenticated session :-/
-   */
   return {
-    addInvolvedStaff: async (req, res) => {
-      if (!res.locals.user.isCoordinator) {
-        throw httpError(401, 'Not authorised to access this resource')
+    viewAddInvolvedStaff: async (req, res) => {
+      const { reportId } = req.params
+
+      const errors = req.flash('errors')
+      const data = { incidentId: reportId }
+
+      res.render('pages/coordinator/add-involved-staff/add-involved-staff.html', { errors, data })
+    },
+
+    submitAddInvolvedStaff: async (req, res) => {
+      const {
+        params: { reportId },
+        body: { username },
+      } = req
+
+      if (!username.trim()) {
+        req.flash('errors', [{ href: '#username', text: "Enter a staff member's username" }])
+        return res.redirect(`/coordinator/report/${reportId}/add-staff`)
       }
 
-      const { reportId, username } = req.params
+      const result = await involvedStaffService.addInvolvedStaff(
+        await systemToken(res.locals.user.username),
+        reportId,
+        username
+      )
 
-      await involvedStaffService.addInvolvedStaff(await systemToken(res.locals.user.username), reportId, username)
-      res.json({ result: 'ok' })
+      req.flash('username', username.toUpperCase())
+      return res.redirect(`/coordinator/report/${reportId}/add-staff/result/${result}`)
     },
 
-    confirmDeleteReport: async (req, res) => {
-      if (!res.locals.user.isCoordinator) {
-        throw httpError(401, 'Not authorised to access this resource')
+    viewAddInvolvedStaffResult: async (req, res) => {
+      const { reportId, result } = req.params
+      const username = firstItem(req.flash('username'))
+
+      const knownResult = Object.values(AddStaffResult).some(knownValue => knownValue === result)
+
+      if (!username || !knownResult || result === AddStaffResult.SUCCESS) {
```

Comment:
As we store the username in flash, we won't have the username after a refresh of this page.
In that case, think it makes sense to just redirect to report view so they can see current state of the users.